datatypes/src/raster/geo_transform.rs

```diff
@@ -64,13 +66,24 @@ impl GeoTransform {
     /// use geoengine_datatypes::primitives::{Coordinate2D};
     ///
     /// let geo_transform = GeoTransform::new_with_coordinate_x_y(0.0, 1.0, 0.0, -1.0);
-    /// assert_eq!(geo_transform.grid_2d_to_coordinate_2d((0, 0)), (0.0, 0.0).into())
+    /// assert_eq!(geo_transform.grid_idx_to_coordinate_2d([0, 0]), (0.0, 0.0).into())
     /// ```
     ///
-    pub fn grid_2d_to_coordinate_2d(&self, grid_index: (usize, usize)) -> Coordinate2D {
-        let (grid_index_y, grid_index_x) = grid_index;
-        let coord_x = self.upper_left_coordinate.x + (grid_index_x as f64) * self.x_pixel_size;
-        let coord_y = self.upper_left_coordinate.y + (grid_index_y as f64) * self.y_pixel_size;
+    #[inline]
+    pub fn grid_idx_to_coordinate_2d(&self, grid_index: GridIdx2D) -> Coordinate2D {
```

Comment:
Why is it necessary to have the unsigned versions? And what about overflows?

Reply:
the unsigned versions are for indexing inside a grid which is more naturally. I guess it is possible to remove them. However, is there a benefit to do this?

Reply:
But is this invariant to the data type? If you have 0 as usize or isize, is the result the same?

Reply:
grid indexing now only uses isize